docs/manual/developer/07_understanding_build_system.md

```diff
@@ -0,0 +1,141 @@
+   checks.
+ - [OCIL](https://csrc.nist.gov/projects/security-content-automation-protocol/specifications/ocil),
+   the Open Checklist Interactive Language, an expressive language for
+   handling manual compliance checks.
+ - [CPE](https://nvd.nist.gov/products/cpe), the Common Platform Enumeration;
+   a scheme for identifying software and systems.
+ - A DS, Datastream format, a mechanism for combining the above into a single
+   redistributable file.
+
+Additionally, some familiarity with the content layout (as discussed in
+previous chapters) is also implied.
+
+However, while this document serves as a guide, ultimately the build system
+is changing and thus inspecting the code is the only way to find the answers
+to many questions.
+
+
+## High-Level Overview
+
+ComplianceAsCode's content project is ultimately the combination of three
+things:
+
+ 1. A collection of content in a format-agnostic manner,
+ 2. A build system for collecting this content and combining it to form
+    artifacts understood by other systems,
+ 3. A test system for validating both the compliance of these artifacts
+    to various standards and the correctness of the content in the repo.
+
+As previous sections describe in detail the expectations around content in the
+repo, this section aims to describe the build system. For understanding of the
+test systems, it is suggested to look at the README under `tests/` in the
+repo.
+
+The build system is generated by CMake and combines local Python utilities
+with XML tools (such as `xmllint` and `xsltproc`) and OpenSCAP's `oscap`
+CLI executable. These Python utilities transform the various input files
+into a more standardized format and apply Jinja macros to them. Ultimately
+many of the artifacts we generate are XML-based so extensive XSLT processing
+occurs after building the initial structure in Python. Finally, OpenSCAP
+combines and references several files for us to build the finished artifacts.
+
+
+### CMake Structure
+
+CMake requires projects have an entry point called `/CMakeLists.txt`. This
+uses the CMake language and drives building and installing the project. This
+file contains several things:
+
+ 1. The many build-time options for customizing the types of content generated,
+ 2. The hand-off for generating each product's specific content,
+ 3. Common installation, testing, and distribution targets.
+
+However, the specifics of building a particular product are contained in the
+shared module located at `cmake/SSGCommon.cmake`. This file contains all of
+the CMake logic to build a particular product and exposes the top-level macro
+`ssg_build_product(...)`. This macro generates per-product build, installation,
+and testing targets. While the specifics should be understood from this file
+directly, in general this takes the following outline of steps in rough order
+of occurrence:
+
+ 1. Generate SCE content and metadata,
+ 2. Generate a shorthand XML from various YAML files (like rules, variables,
```

Comment:
shorthand is a term that appears here without prior explanation in this document

Reply:
I ended up writing this after I wrote the CMake documentation. I don't think shorthand is ever defined elsewhere either and is kinda in flux as the build process changes. Do you have a suggestion around how to introduce it? It isn't a standard we can link to externally, right?

Reply:
It isn't a standard, it's our invention and unfortunately the format isn't documented anywhere. I think you have introduced it nicely in the code comments: "a single XML document that resembles the structure of the XCCDF but isn't standards compliant", you can use it here as well.

Reply:
@jan-cerny updated :)